components/eam/src/physics/cam/gw_common.F90

```diff
@@ -742,4 +746,1472 @@ subroutine gw_drag_prof(ncol, ngwv, src_level, tend_level, do_taper, dt, &
+  real(r8), dimension( ims:ime,kms:kme ),    intent(in) ::  prsl
+  real(r8), dimension( ims:ime,kms:kme ),    intent(in) ::  prslk
+  real(r8), dimension( ims:ime,kms:kme+1),   intent(in) ::  prsi
+  real(r8), dimension( ims:ime,kms:kme ),    intent(in) ::  del
+  real(r8), dimension( ims:ime,kms:kme ),    intent(in) ::  bnv_in
+  !input pbl level index
+  integer,dimension( ims:ime ),              intent(in) ::  kpbl
+  integer,                                   intent(in) ::  kpblmax
+  !input for small-scale ogwd
+  real(r8), dimension( ims:ime ),            intent(in) :: br1
+  real(r8), dimension( ims:ime ),            intent(in) :: hpbl
+  real(r8), dimension( ims:ime ),            intent(in) :: xland1
+  real(r8), dimension( ims:ime, kms:kme ),   intent(in) :: dz2
+  !variables for open/close process
+  logical,                                   intent(in) ::  gsd_gwd_ls
+  logical,                                   intent(in) ::  gsd_gwd_bl
+  logical,                                   intent(in) ::  gsd_gwd_ss
+  logical,                                   intent(in) ::  gsd_gwd_fd
+  !input/output tendencies
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dudt
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dvdt
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dthdt
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtaux2d_ls
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtauy2d_ls
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtaux2d_bl
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtauy2d_bl
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtaux2d_ss
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtauy2d_ss
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtaux2d_fd
+  real(r8), dimension( ims:ime,kms:kme ),    intent(inout) ::  dtauy2d_fd
+  !input/output total column stress
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dusfc_ls
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dvsfc_ls              
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dusfc_bl
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dvsfc_bl              
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dusfc_ss
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dvsfc_ss              
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dusfc_fd
+  real(r8), dimension( ims:ime )        ,    intent(inout) ::  dvsfc_fd
+  !
+  ! added for small-scale orographic wave drag
+  !
+  integer                                  :: kpbl2,kvar
+  real(r8), dimension(its:ite,kts:kte)     :: utendwave,vtendwave,thx,thvx,za
+  real(r8), dimension(its:ite)             :: govrth
+  real(r8), dimension(its:ite,kts:kte+1)   :: zq
+  real(r8)                                 :: tauwavex0,tauwavey0,XNBV,density,tvcon,hpbl2
```

Comment:
XNBV is a pretty non-descriptive variable name. Looking below it seems to be a temporary work variable - could we use a name to communicate that better?